polkadot/node/core/pvf/common/src/error.rs

```diff
@@ -32,35 +31,43 @@ pub type PrecheckResult = Result<(), PrepareError>;
 
 /// An error that occurred during the prepare part of the PVF pipeline.
 // Codec indexes are intended to stabilize pre-encoded payloads (see `OOM_PAYLOAD`)
-#[derive(Debug, Clone, Encode, Decode)]
+#[derive(thiserror::Error, Debug, Clone, Encode, Decode)]
 pub enum PrepareError {
 	/// During the prevalidation stage of preparation an issue was found with the PVF.
 	#[codec(index = 0)]
+	#[error("prevalidation: {0}")]
 	Prevalidation(String),
 	/// Compilation failed for the given PVF.
 	#[codec(index = 1)]
+	#[error("preparation: {0}")]
 	Preparation(String),
 	/// Instantiation of the WASM module instance failed.
 	#[codec(index = 2)]
+	#[error("runtime construction: {0}")]
 	RuntimeConstruction(String),
 	/// An unexpected error has occurred in the preparation job.
 	#[codec(index = 3)]
+	#[error("panic: {0}")]
 	JobError(String),
```

Comment:
```suggestion
	#[error("prepare job error: {0}")]
	JobError(String),
```

(That's my bad, I forgot to update this line after renaming `Panic` -> `JobError`.

Reply:
For sure:) I've noticed it but from [the book description](https://paritytech.github.io/polkadot-sdk/book/node/utility/pvf-host-and-workers.html) it was quite logical

Reply:
For consistency we could also make a `prepare: ` prefix for other messages in `PrepareError` if it is allowed

Reply:
> I'm interested in whether we can now better log some of these errors that didn't implement the Display trait before. We should be able to log them with {} instead of {:?}. I'll look through the codebase later for such instances.

It is good point! Thank you! I will check these instances within the PR

Reply:
@mrcnski I've checked the usage of `{:?}` within `pvf/` and changed several ones to `{}`. And also changed `PrepareError` messages a little bit for consistency.